lib/web_of_science/process_records.rb

```diff
@@ -33,49 +41,68 @@ def execute
 
       delegate :links_client, to: :WebOfScience
 
-      # @return [Array<String>] WosUIDs that successfully create a new Publication or Contribution
+      # from the incoming (db-filtered) records
+      def uids
+        @uids ||= records.map(&:uid)
+      end
+
+      # @return [Array<String>] WosUIDs that successfully create a new Publication
       def create_publications
         return [] if records.empty?
-        matching_uids = Publication.where(wos_uid: records.map(&:uid)).pluck(:wos_uid)
-        save_wos_records(records.reject { |rec| matching_uids.include? rec.uid })
-        records.select { |rec| !matching_contribution(author, rec) && create_publication(rec) }
-               .map(&:uid)
-               .uniq
+        wssrs = save_wos_records
+        wssrs_hash = wssrs.map(&:uid).zip(wssrs).to_h
```

Comment:
not sure i understand what is happening here - would you mind just walking me through the logic (in person/zoom if more efficient)?